data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -102,13 +104,19 @@ public static List<DataFile> listPartition(
 
       Path partition = new Path(partitionUri);
       FileSystem fs = partition.getFileSystem(conf);
-      List<FileStatus> fileStatus =
-          Arrays.stream(fs.listStatus(partition, HIDDEN_PATH_FILTER))
-              .filter(FileStatus::isFile)
-              .collect(Collectors.toList());
-      DataFile[] datafiles = new DataFile[fileStatus.size()];
+      FileStatus[] fileStatuses = fs.listStatus(partition, HIDDEN_PATH_FILTER);
+      List<FileStatus> files = Lists.newArrayList();
```

Comment:
nit, maybe leave List with original name `fileStatuses` and get a new name for array to avoid all those rename change below? I think Array is only used once here